hazelcast/src/test/java/com/hazelcast/spi/impl/sequence/FailFastCallIdSequenceTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.sequence;
+
+import com.hazelcast.core.HazelcastOverloadException;
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.impl.operationservice.impl.DummyBackupAwareOperation;
+import com.hazelcast.spi.impl.operationservice.impl.DummyOperation;
+import com.hazelcast.spi.impl.operationservice.impl.DummyPriorityOperation;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.RequireAssertEnabled;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.CountDownLatch;
+
+import static com.hazelcast.spi.OperationAccessor.setCallId;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class FailFastCallIdSequenceTest extends HazelcastTestSupport {
+
+    FailFastCallIdSequence sequence = new FailFastCallIdSequence(100);
+
+    @Test
+    public void test() {
+        assertEquals(0, sequence.getLastCallId());
+        assertEquals(100, sequence.getMaxConcurrentInvocations());
+    }
+
+    @Test
+    public void whenNext_thenSequenceIncrements() {
+        // regular operation
+        testNext(new DummyOperation());
+
+        // backup-aware operation
+        testNext(new DummyBackupAwareOperation());
+
+        // urgent operation
+        testNext(new DummyPriorityOperation());
+    }
+
+    private void testNext(Operation operation) {
+        long oldSequence = sequence.getLastCallId();
+        long result = nextCallId(sequence, operation.isUrgent());
+        assertEquals(oldSequence + 1, result);
+        assertEquals(oldSequence + 1, sequence.getLastCallId());
+    }
+
+    @Test (expected = HazelcastOverloadException.class)
+    public void next_whenNoCapacity_thenThrowException() throws InterruptedException {
+        sequence = new FailFastCallIdSequence(1);
+        final long oldLastCallId = sequence.getLastCallId();
+
+        final CountDownLatch nextCalledLatch = new CountDownLatch(1);
```

Comment:
You don't need to spawn a task for it.

If you would do:
```
// take the only slot available
callIdSequence.next()

// this next is going to fail with an exception
callIdSequence.next()

```

Reply:
OK, changed.